### PR TITLE
Add custom hive catalog to not override existing Hive metadata

### PR DESCRIPTION
**Pull request by shardulm94**

### CODE REVIEW

File: hive/src/main/java/org/apache/iceberg/hive/CustomHiveCatalog.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+
+
+/**
+ * A {@link HiveCatalog} which uses {@link CustomHiveTableOperations} underneath.
+ *
+ * This catalog is especially useful if the Iceberg operations are being performed in response to Hive operations
+ */
+public class CustomHiveCatalog extends HiveCatalog {
```
Comment (rdsr) [inline hive/src/main/java/org/apache/iceberg/hive/CustomHiveCatalog.java]: Do you want to add an entry to `org.apache.iceberg.hive.HiveCatalogs` for caching custom catalogs?

Comment (shardulm94) [reply hive/src/main/java/org/apache/iceberg/hive/CustomHiveCatalog.java]: Done

File: hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
```
Comment (rdsr) [inline hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java]: nit: Test files name Test as prefix not suffix

File: hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      Assert.assertEquals(2, hiveTable.getSd().getCols().size());
+      Assert.assertFalse(hiveTable.getParameters().containsKey("testProp"));
+      Assert.assertEquals("org.apache.hadoop.hive.ql.io.orc.OrcSerde",
+          hiveTable.getSd().getSerdeInfo().getSerializationLib());
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Table createHiveTable() {
```
Comment (rdsr) [inline hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java]: nit: rename this to hiveTable() only?

Comment (shardulm94) [reply hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java]: Fixed

File: hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java (hunk 2 above)
Comment (shardulm94) [reply hive/src/test/java/org/apache/iceberg/hive/CustomHiveCatalogTest.java]: Fixed

File: hive/src/main/java/org/apache/iceberg/hive/CustomHiveTableOperations.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    } catch (TException e) {
+      String errMsg = String.format("Failed to get table info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
```
Comment (rdsr) [inline hive/src/main/java/org/apache/iceberg/hive/CustomHiveTableOperations.java]: Seems like `refreshFromMetadataLocation` can be called with a null metadataLocation. Should we move this in the else block?

**Pull request merged**